api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -122,6 +122,8 @@ func (mw *azureManagedControlPlaneWebhook) ValidateCreate(ctx context.Context, o
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
+//
+//nolint:gocyclo // This functions requires many checks to validate each field of the amcp.
```

Comment:
Would it be possible or worthwhile to refactor this function as a big loop that calls `ValidateImmutable`? It might be more readable and could silence the `gocyclo` linter.